cmd/object-handlers.go

```diff
@@ -298,50 +285,57 @@ func (api objectAPIHandlers) CopyObjectHandler(w http.ResponseWriter, r *http.Re
 
 	/// maximum Upload size for object in a single CopyObject operation.
 	if isMaxObjectSize(objInfo.Size) {
-		writeErrorResponse(w, r, ErrEntityTooLarge, objectSource)
-		return
-	}
-
-	// Size of object.
-	size := objInfo.Size
-
-	pipeReader, pipeWriter := io.Pipe()
-	go func() {
-		startOffset := int64(0) // Read the whole file.
-		// Get the object.
-		gErr := objectAPI.GetObject(sourceBucket, sourceObject, startOffset, size, pipeWriter)
-		if gErr != nil {
-			errorIf(gErr, "Unable to read an object.")
-			pipeWriter.CloseWithError(gErr)
+		writeErrorResponse(w, r, ErrEntityTooLarge, cpSrcPath)
+		return
+	}
+
+	cpMetadataOnly := strings.EqualFold(cpSrcPath, cpDestPath)
+	var newMetadata map[string]string
+	// Source and destination objects cannot be same if x-amz-metadata-directive is not set.
```

Comment:
this comment is not correct because, src and dst can't be same even if x-amz-metadata-directive is set to COPY